Compare charts over location/time periods; COUNTRY=cambodia

laurentS opened the pull request:
This fixes issue #930 and #931 

How to test the feature:

- go to the charts tab
- click the "compare locations" switch
- select locations 1 & 2
- select time period and chart layers
- see charts show up

## TODO
- [x] Provide 2 CSV files when comparing 
- [x] Limit number of selectable layers to 2 (is this really needed?)
- [x] Adjust styling
- [x] Add time period comparison


Screenshot/video of feature:
<img width="1727" alt="Screenshot 2023-10-23 at 5 18 38 PM" src="https://github.com/WFP-VAM/prism-app/assets/16843267/9f7a9dd9-16d7-42f7-af7c-c913986e30b2">

laurentS commented:
This is more or less ready to review. There's an open point in the todo list above. I think it makes sense to keep the code as is, allowing multiple layers to be compared, but maybe I'm missing something on the "business" logic side.

Also CSV filenames are a bit chaotic, not sure how to make them nicer. Ideas welcome!

wadhwamatic commented:
This is looking great, thanks @laurentS! I would like to test a bit more locally tomorrow with other regions but the surge deployment mostly works as expected. I did see strange behavior with the CSV filenames. It seems an admin 2 name is being added to the filename even if I am only requesting to admin 1. See screenshots below.

<img width="504" alt="Screenshot 2023-10-23 at 18 48 34" src="https://github.com/WFP-VAM/prism-app/assets/3343536/f347a4f4-2cc6-42b6-ac1f-c7970e835fb9">
<img width="679" alt="Screenshot 2023-10-23 at 18 51 14" src="https://github.com/WFP-VAM/prism-app/assets/3343536/ae7225a2-00fc-45cf-aeb2-63e9b5020232">
<img width="502" alt="Screenshot 2023-10-23 at 18 45 49" src="https://github.com/WFP-VAM/prism-app/assets/3343536/1f16c5b6-6c6c-446a-91ca-c83ae9fc80a7">
<img width="746" alt="Screenshot 2023-10-23 at 18 45 41" src="https://github.com/WFP-VAM/prism-app/assets/3343536/a7e7baed-388d-468b-8c77-6a96a10863f4">

I'm also thinking about how we can improve the UX for time period selection. More on this tomorrow!

wadhwamatic commented:
Comparing locations when we have admin0 (country-level) data crashes the app. Screenshot below.

Steps to reproduce:
1. COUNTRY=rbd
2. Go to charts choose a single location (country, Benin for example)
3. Keep default dates
4. Choose any chart
5. Next, toggle 'compare locations'. The app will crash with the message below. This seems to only happen if you choose a chart for a single country and then try to compare locations afterwards 

<img width="1109" alt="Screenshot 2023-10-24 at 17 25 40" src="https://github.com/WFP-VAM/prism-app/assets/3343536/759797e4-b2ec-41d7-8714-de9522981f55">

wadhwamatic commented:
When comparing locations for the RBD deployment, we don't need the text 'Regional Bureau Dakar' in the location header. It incorrectly replaces the country name. See screenshots below.

<img width="1405" alt="Screenshot 2023-10-24 at 17 49 55" src="https://github.com/WFP-VAM/prism-app/assets/3343536/9fa5b805-2a39-44da-b903-7ec7b4025140">

<img width="1415" alt="Screenshot 2023-10-24 at 17 50 33" src="https://github.com/WFP-VAM/prism-app/assets/3343536/32a6ff14-eb79-493a-baa0-4c52dd8eaa3b">

laurentS commented:
> Comparing locations when we have admin0 (country-level) data crashes the app. Screenshot below.

Fixed in 88c70e5172d70f140c53e091929ac8a75f8195be

> When comparing locations for the RBD deployment, we don't need the text 'Regional Bureau Dakar' in the location header.

Fixed in ca3b242a8df7b0b2cb63f5c73d3b022a3b0fcc66

> It seems an admin 2 name is being added to the filename even if I am only requesting to admin 1.

Fixed in 8445cd2695cec00567ed95e9b8ddb8a40932d19f

I think I got all your comments above.

As a side note, loading the charts page is extremely slow on my machine (1 year old laptop, reasonably powerful) on a 1Gb/s fiber internet connection. It seems that with country=rbd, there is something like 160MB of country boundary data that is loaded. Most of it is cached, but the page still takes over 12 seconds to be responsive, which seems like too much to me. Not sure if we can do something about it?

ericboucher commented on frontend/src/components/MapView/LeftPanel/ChartsPanel/index.tsx:
```diff
@@ -264,157 +261,143 @@ const ChartsPanel = memo(
-      },
-      [selectedAdmin1Area, data],
-    );
+    const generateSecondCSVFilename = useCallback(() => {
```
Can we find a better name than"SecondCSVFilename"?

wadhwamatic reviewed (approved):
Thanks for the fixes @laurentS. All looks good from my side now.

Regarding the performance issues, I don't see it so badly on my side but this has been a known issue. See https://github.com/WFP-VAM/prism-app/issues/467. cc @ericboucher
